pyspecdata/core.py

```diff
@@ -7255,25 +7209,26 @@ def rename(self,previous,new):
             self.fit_axis = new
         nddata.rename(self,previous,new)
         return self
-    def fit(self,set = None, set_to = None, force_analytical = False, silent = False):
+    def fit(self,set_what = None, set_to = None, force_analytical = False, silent = False):
         r'''actually run the fit'''
         if not silent: print("\n")
         if not silent: print(r'\resizebox*{!}{3in}{\begin{minipage}{\linewidth}')
-        if isinstance(set, dict):
-            set_to = list(set.values())
-            set = list(set.keys())
+        if isinstance(set_what, dict):
+            set_to = list(set_what.values())
+            set_what = list(set_what.keys())
         x = self.getaxis(self.fit_axis)
         if iscomplex(self.data.flatten()[0]):
-            if not silent: print(lsafen('Warning, taking only real part of fitting data!'))
+            if not silent: print((lsafen('Warning, taking only real part of fitting data!')))
         y = real(self.data)
         sigma = self.get_error()
         if sigma is None:
             if not silent: print('{\\bf Warning:} You have no error associated with your plot, and I want to flag this for now\n\n')
             warnings.warn('You have no error associated with your plot, and I want to flag this for now',Warning)
             sigma = ones(shape(y))
-        p_ini = real(array(self.guess())) # need the numpy format to allow boolean mask
-        if set != None:
-            self.set_indices,self.set_to,self.active_mask = self.gen_indices(set,set_to)
+        p_ini = [1.0,1.0,1.0] # hard-coded for debug
```

Comment:
here, I would say that this is a fine default guess, but rather than hard-coding it like this, we should allow the user to set a dictionary that gives the initial guess variables.  The code actually won't work for anything that has  more than 3 parameters without doing this.

If this requires further discussion, I would set this to be `[1.0]*number_of_parameters` (where `number_of_parameters` is the appropriate variable for the number of parameters, and we can address this in a subsequent pull request

Reply:
I agree this hard-coded approach is not the way to go moving forward. I like the number_of_parameters idea.

Reply:
Alright, so this worked by just plugging in number of parameters as 3 -- now just need to find a way to get it to pull the number of parameters from what is provided by the user

Reply:
OK, so I define a new variable self.number_of_parameters that pulls from functional_form when we are setting up the expression, and this appears to work (without hard-coding number_of_parameters as 3)

Reply:
this sounds like a good solution

Reply:
I deal with this in a new pull request.